src/modules/avformat/producer_avformat.c

```diff
@@ -1450,6 +1461,21 @@ static int convert_image( producer_avformat self, AVFrame *frame, uint8_t *buffe
 			out_data, out_stride);
 		sws_freeContext( context );
 	}
+	else if ( *format == mlt_image_yuv422p16 )
+	{
+		int flags = mlt_get_sws_flags(width, height, src_pix_fmt, width, height, AV_PIX_FMT_RGBA);
+		struct SwsContext *context = sws_getContext( width, height, src_pix_fmt,
+			width, height, AV_PIX_FMT_YUV422P16LE, flags, NULL, NULL, NULL);
+		uint8_t *out_data[8];
+		int out_stride[8];
+		av_image_fill_arrays(out_data, out_stride, buffer, AV_PIX_FMT_YUV422P16LE, width, height, IMAGE_ALIGN);
+		// libswscale wants the RGB colorspace to be SWS_CS_DEFAULT, which is = SWS_CS_ITU601.
+		mlt_set_luma_transfer( context, self->yuv_colorspace, 2100, self->full_luma, 0 );
```

Comment:
I should also point out here that I optimistically chose 2100 here even though I don't know if MLT supports it.